requests/src/main/scala/weco/api/requests/services/RequestsService.scala

```diff
@@ -0,0 +1,80 @@
+package weco.api.requests.services
+
+import grizzled.slf4j.Logging
+import weco.api.search.elasticsearch.ElasticsearchError
+import weco.api.stacks.models.HoldRejected.ItemUnavailableFromSourceSystem
+import weco.api.stacks.models.{HoldAccepted, HoldRejected}
+import weco.api.stacks.services.SierraService
+import weco.catalogue.internal_model.identifiers.{
+  CanonicalId,
+  IdState,
+  SourceIdentifier
+}
+import weco.catalogue.internal_model.identifiers.IdentifierType.SierraSystemNumber
+import weco.catalogue.internal_model.work.Item
+import weco.sierra.models.identifiers.SierraPatronNumber
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class RequestsService(
+  sierraService: SierraService,
+  itemLookup: ItemLookup
+)(implicit executionContext: ExecutionContext)
+    extends Logging {
+
+  def makeRequest(
+    itemId: CanonicalId,
+    patronNumber: SierraPatronNumber
+  ): Future[Either[HoldRejected, HoldAccepted]] = {
+    itemLookup.byCanonicalId(itemId).flatMap {
+      case Right(
+          Item(
+            IdState.Identified(
+              _,
+              srcId @ SourceIdentifier(SierraSystemNumber, _, _),
+              _
+            ),
+            _,
+            _,
+            _
+          )
+          ) =>
+        sierraService.placeHold(
+          patron = patronNumber,
+          sourceIdentifier = srcId
+        )
+
+      case Right(sourceIdentifier) =>
+        warn(s"Cannot request from source: $itemId / $sourceIdentifier")
+        Future.successful(Left(ItemUnavailableFromSourceSystem))
```

Comment:
These right arms look like they have completely different types. Is this right? When would we get any sort of source identifier from itemLookup?

Reply:
The autoformatting has unhelpfully rearranged this. It's pattern matching on the `IdentifierType` in `SourceIdentifier`, the same behaviour this logic had before - i'll put the `if` back as it is definitely clearer.